src/appleseed.python/bindblendermeshconverter.cpp

```diff
@@ -0,0 +1,219 @@
+                    vert.no[1],
+                    vert.no[2]));
+        }
+    }
+
+    int uv_index = 0;
+
+    // Push triangles
+    for (size_t tri_index = 0; tri_index < triangle_length; ++tri_index)
+    {
+        MeshFace &face = triangles[tri_index];
+        blender_mesh->push_triangle(
+            Triangle(
+                face.v[0],
+                face.v[1],
+                face.v[2],
+                face.mat_nr));
+
+        if (export_normals == true || export_uvs == true)
+        {
+            Triangle& tri = blender_mesh->get_triangle(tri_index);
+
+            // Tie vertex normals to mesh faces
+            if (export_normals == true)
+            {
+                tri.m_n0 = face.v[0];
+                tri.m_n1 = face.v[1];
+                tri.m_n2 = face.v[2];
+            }
+
+            // Tie uv coordinates to mesh faces
+            if (export_uvs == true)
+            {
+                MeshTexFace &tex_face = uv_face[tri_index];
+
+                blender_mesh->push_tex_coords(
+                    GVector2(
+                        tex_face.uv[0][0],
+                        tex_face.uv[0][1]));
+                tri.m_a0 = uv_index++;
+
+                blender_mesh->push_tex_coords(
+                    GVector2(
+                        tex_face.uv[1][0],
+                        tex_face.uv[1][1]));
+                tri.m_a1 = uv_index++;
+
+                blender_mesh->push_tex_coords(
+                    GVector2(
+                        tex_face.uv[2][0],
+                        tex_face.uv[2][1]));
+                tri.m_a2 = uv_index++;
+            }
+        }
+    }
+}
+
+void convert_vertex_pose(
+    MeshObject*                 blender_mesh,
+    const size_t                pose,
+    const size_t                vertices_length,
+    const size_t                vertex_pointer,
+    const size_t                triangle_length,
+    const size_t                triangle_pointer,
+    const bool                  export_normals)
+{
+    // Convert size_t numbers to actual pointers
+    MeshVert* vertices = reinterpret_cast<MeshVert *>(vertex_pointer);
+    MeshFace* triangles = reinterpret_cast<MeshFace *>(triangle_pointer);
+
+    // Push vertices
+    for (size_t vertex_index = 0; vertex_index < vertices_length; ++vertex_index)
+    {
+        MeshVert &vert = vertices[vertex_index];
+        blender_mesh->set_vertex_pose(
+            vertex_index,
+            pose,
+            GVector3(
+                vert.co[0],
+                vert.co[1],
+                vert.co[2]));
+
+        // Push normals
+        if (export_normals == true)
+        {
+            blender_mesh->set_vertex_normal_pose(
+                vertex_index,
+                pose,
+                GVector3(
+                    vert.no[0],
+                    vert.no[1],
+                    vert.no[2]));
+        }
+    }
+}
+
+void bind_blender_mesh_converter()
+{
+    bpy::def("convert_mesh", &convert_mesh);
```

Comment:
Maybe make python names more specific: bl_convert_mesh() or convert_blender_mesh()?
I think convert_mesh is too generic.

Reply:
Changed.